tutorials/preprocessing/35_artifact_correction_regression.py

```diff
@@ -23,146 +24,195 @@
 Other approaches like :ref:`ICA <tut-artifact-ica>` or
 :ref:`SSP <tut-artifact-ssp>` will likely work better for ECG.
 
+Furthermore, regression approaches are usually performed in situations where
+there are few channels available, and removing an entire signal component is
+undesirable. Hence, most articles on the topic concern EEG and it is
+unusual to see the technique applied to MEG. For this reason, we will restrict
+the analysis in this tutorial to EEG data only.
+
+
 Prepare the data
 ^^^^^^^^^^^^^^^^
 
 We begin as always by importing the necessary Python modules and loading some
-data. In this case we use data from :ref:`Brainstorm <tut-brainstorm-auditory>`
-because it has some clear, large blink artifacts. We then crop it to 60
-seconds, set some channel information, then process the auditory evoked data.
-
-Note that there are other corrections that are useful for this dataset that we
-will not apply here (see :ref:`tut-brainstorm-auditory` for more information).
+data. The :ref:`MNE-Sample <sample-dataset>` dataset has some clear, large
+blink artifacts, especially during the presentation of visual stimuli.
 """
 
 # %%
-
-import os.path as op
 import numpy as np
 import mne
+from mne.preprocessing import EOGRegression
+
+data_path = mne.datasets.sample.data_path()
+raw_fname = data_path / 'MEG' / 'sample' / 'sample_audvis_raw.fif'
+raw = mne.io.read_raw_fif(raw_fname)
+raw.pick(['eeg', 'eog', 'stim'])
+raw.load_data()
+
+# The regression technique works regardless of chosen reference. However, it is
+# important to choose a reference before proceeding with the analysis.
+raw.set_eeg_reference('average')
+raw.apply_proj()
```

Comment:
This is not necessary and a no-op here. The average reference is applied in the previous line.

Reply:
if proj=True is not applied yet I think.

Reply:
There are no projectors available (it also says so in the output).

Reply:
projection=False is the default in raw.set_eeg_reference so yes. Now you may have other projections present in the file?

Reply:
But not in the example.